app/code/Magento/Customer/Model/ResourceModel/AddressRepository.php

```diff
@@ -139,7 +139,17 @@ public function save(\Magento\Customer\Api\Data\AddressInterface $address)
         $this->addressRegistry->push($addressModel);
         $this->updateAddressCollection($customerModel, $addressModel);
 
-        return $addressModel->getDataModel();
+        $defaultBillingAddressId = null;
+        if ($customerModel->getDefaultBillingAddress() instanceof CustomerAddressModel) {
+            $defaultBillingAddressId = $customerModel->getDefaultBillingAddress()->getId();
+        }
+
+        $defaultShippingAddressId = null;
+        if ($customerModel->getDefaultShippingAddress() instanceof CustomerAddressModel) {
```

Comment:
Magento doesn't encourage usage of instanceof operator, as this couldn't be considered as good OOP practice.
taking into account that 
`
     * @return  Address|false
     */
    public function getPrimaryAddress($attributeCode)
`
Change your conditional logic to this presentation:

`
if ($customerModel->getDefaultBillingAddress()) {
`

Also this will introduce micro optimization